backend/src/Squidex/Config/Authentication/OAuth2Handler.cs

```diff
@@ -0,0 +1,67 @@
+﻿// ==========================================================================
+//  Squidex Headless CMS
+// ==========================================================================
+//  Copyright (c) Squidex UG (haftungsbeschränkt)
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Newtonsoft.Json.Linq;
+
+namespace Squidex.Config.Authentication
+{
+    public sealed class OAuth2Handler : OAuthEvents
+    {
+        public override Task RedirectToAuthorizationEndpoint(RedirectContext<OAuthOptions> context)
+        {
+            context.Response.Redirect(context.RedirectUri);
+
+            return Task.CompletedTask;
+        }
+
+        public override Task TicketReceived(TicketReceivedContext context)
+        {
+            var test = context.Request.Body;
+            return base.TicketReceived(context);
+        }
+
+        public override Task RemoteFailure(RemoteFailureContext context)
+        {
+            return base.RemoteFailure(context);
+        }
+
+        public override Task AccessDenied(AccessDeniedContext context)
+        {
+            return base.AccessDenied(context);
+        }
+
+        public override async Task CreatingTicket(OAuthCreatingTicketContext context)
+        {
+            if (context.Options.UserInformationEndpoint != null)
```

Comment:
What is this? Is this part of the oauth specification? If yes, why is not handled by the default handler?

Reply:
It's a part of oauth specification if there's a REST service to retrieve user informations. Unfortunately it's not handled by the default handler.

Reply:
In OpenID connect the user id claim is called "sub" ... is this no available in the token by default?

Reply:
I like this approach with the claim actions: https://developer.okta.com/blog/2019/07/12/secure-your-aspnet-core-app-with-oauth